### ISSUE TITLE
[ChipGroup] check function not working in JellyBean (API 16)

### ISSUE THREAD

**Issue by GustavoAraujoLetras** [material-components/material-components-android#1493]
**Description:** In a JellyBean device, if some Chips were created with no id (`View.NO_ID`) and added to the ChipGroup, the group is unable to set chips as checked/unchecked or guarantee that only one chip is selected at any given moment.

**Expected behavior:** ChipGroup should be able to perform `check(Int)` and handle single selection as described on the documentation.

**Source code:**
```kotlin
val chipGroup = ChipGroup(context).apply {
    isSingleLine = true
    isSelectionRequired = true
    isSingleSelection = true
}

val firstChip = Chip(context).apply { text = "All" }
chipGroup.addView(it)

listOfStrings.map {
    Chip(context).apply { text = it }
}.forEach {
    chipGroup.addView(it)
}

chipGroup.check(firstChip) // nothing is selected

// User can also click in all chips and all of them will behave independently
```

**Android API version:** 16

**Material Library version:** 1.2.0-rc01

**Device:** Emulator

**Comment by matpag** [comment]
You can't expect the correct behaviour without passing a unique ID to the `Chip` object.
`ChipGroup` uses the view id to identify a single selection.

Why are you using `View.NO_ID` and not `View.generateViewId()?`

**Comment by matpag** [comment]
I've found the "missing" part.
on API < 17 the method `View.setId(int id)` was [this](https://cs.android.com/android/platform/superproject/+/android-4.0.4_r2.1:frameworks/base/core/java/android/view/View.java;l=12356-12358) while in API >= 17 is [this](https://cs.android.com/android/platform/superproject/+/android-4.3.1_r1:frameworks/base/core/java/android/view/View.java;l=15502-15507).

In practice the Android framework started generating unique IDs when you call `setId` or `setLabelFor`.
I think with those information should be easy to understand when those methods are called

**Comment by GustavoAraujoLetras** [comment]
When I was still debugging it, I realize that the "not setting as checked" part is caused by the `findViewById` used [here](https://github.com/material-components/material-components-android/blob/f945a0649b9cddc13afafeb80df7218f1aa85ef8/lib/java/com/google/android/material/chip/ChipGroup.java#L372-L379) returning `null`. That was enough for me to change my approach. I don't have that much knowledge to deep dive into the View source-code and I didn't want to take any longer to finish my issue (in my project).

Could be a weird behavior when using the `hashCode` as an ID but honestly that was as far as I went giving a thought about it before throwing it away.
I implemented a ChipGroupCompat on my project as a workaround and, instead of using the View ID I used the Chip instance itself. It worked pretty well, but I didn't tested as much as it should to be published because I end up not using it.

I have some uncertainties about how stable my code is because I'm not used to create components that must be as generic as this components, but I'm pretty sure that there's not much loss doing the way I did. My main concern about my approach about saved states, but I haven't tested this. I'll might test it later today.
